third-party/hwloc/Makefile

```diff
@@ -11,6 +11,9 @@ include $(CHPL_MAKE_HOME)/make/Makefile.base
 ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
 CHPL_HWLOC_CFG_OPTIONS += --host=x86_64-cle-linux-gnu
 endif
+ifneq ($(CHPL_MAKE_HOST_PLATFORM),$(CHPL_MAKE_TARGET_PLATFORM)))
```

Comment:
Misc things caught by nightly whitebox testing:

There's an extra `)` here (looks like it's only for the hwloc makefile though, the others look good.)

Also, I think this should be an `else ifneq`. For whitebox testing `CHPL_HOST_PLATFORM=linux64` and `CHPL_TARGET_PLATFORM=cray-xc`, which should just trigger the above cross compile for `cray-x` , but its instead triggering both and the second one takes precedence.